src/components/web3-connect/index.tsx

```diff
@@ -122,8 +141,13 @@ export class Container extends React.Component<Properties, State> {
       this.syncGlobalsForConnectedStatus();
     }
 
+    if (!web3.account && web3.account !== previouslyAccount && !web3.active && !web3.library) {
```

Comment:
can we just base this on the connection? do we need all of the other checks? eg. i think if the status goes from connected -> disconnected, we should reconnect infura. not sure if that just means checking web3.active, or if it'll be a bit more involved.

Reply:
We can't just base on the connection, cuz the status here doesn't goes from connected to disconnected, and to guaranteed that the wallet was disconnected we should checking the web3.account .